File: scripts/algorithms/M/Minimum Operations to Make the Array Increasing/Minimum Operations to Make the Array Increasing.js
var minOperations = function(nums) {
    if(nums.length < 2) return 0;
    let count = 0;
    for(let i = 1; i<nums.length; i++) {
       if(nums[i] <= nums[i-1]) {
           let change = nums[i-1] - nums[i] + 1;
           count += change;
           nums[i] += change;
       }
    }
    
    return count;
};

File: scripts/algorithms/M/Minimum Operations to Make the Array Increasing/Minimum Operations to Make the Array Increasing.cpp
// Runtime: 12 ms (Top 95.79%) | Memory: 15.9 MB (Top 34.27%)
class Solution {
public:
    int minOperations(vector<int>& nums) {
        int output=0;
        for(int i=0;i<nums.size()-1;i++){
            if(nums[i]<nums[i+1])
                continue;
            else{
                output=output+(nums[i]+1-nums[i+1]);
                nums[i+1]=nums[i]+1;
            }
        }
        return output;
    }
};